src/rgw/rgw_rados.cc

```diff
@@ -12791,7 +12791,7 @@ int RGWRados::cls_bucket_list(RGWBucketInfo& bucket_info, int shard_id, rgw_obj_
           return r;
       }
     }
-    if (r >= 0) {
+    if (r >= 0 && start.name != dirent.key.name) {
```

Comment:
@joke-lee this fixes the result set, but should the key name have already been filtered at the OSD?  (I'm asking sincerely, I suspect but don't know that is the case yet)

Reply:
hi @mattbenjamin ,i have update the code, just like 
```
if (!params.list_versions && !entry.is_visible()) 
``` the code filter delete_marker and current version object, i think filter the object here is better